### PR TITLE
Add ErrorBadgeManager and UI for badging Screen tabs

### PR DESCRIPTION
**Pull request by kenzieschmoll**
This CL adds the skeleton for adding error-badging to devtools tabs. 

This is a checkpoint with the base support, and error handling for appropriate screens will be done in follow up PRs. The result will look like this:
Screen Shot 2020-12-15 at 4 35 23 PM
Screen Shot 2020-12-15 at 4 35 27 PM

### CODE REVIEW

**Comment by devoncarew** [comment]
As a drive by comment on the look, did you try badging the lower right of the tab text? It might be alright to overlap the tab title (it looks good as is, but I was curious about how other options might look).

File: packages/devtools_app/lib/src/error_badge_manager.dart [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    // Log stderr events.
+    autoDispose(service.onStderrEvent.listen(_handleStdErr));
+  }
+
+  void _handleExtensionEvent(Event e) async {
+    // TODO(kenz): handle Flutter.Error extension events.
+  }
+
+  void _handleStdErr(Event e) {
+    // TODO(kenz): handle stderr events
```
Comment (jacob314) [inline packages/devtools_app/lib/src/error_badge_manager.dart:39]: might want to highlight the logging page when these show up?

File: packages/devtools_app/lib/src/error_badge_manager.dart [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  }
+
+  ValueListenable<int> errorCountNotifier(String screenId) {
+    return _activeErrorCounts[screenId] ?? ValueNotifier<int>(0);
```
Comment (jacob314) [inline packages/devtools_app/lib/src/error_badge_manager.dart:43]: I think ben added a class that is a ValueNotifier with a const Value. You'll want that for the case where the value is zero to make it a little clearer what is going on.

**Review by jacob314 — APPROVED** [review]
lgtm

File: packages/devtools_app/lib/src/error_badge_manager.dart (hunk 2 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/error_badge_manager.dart:43]: @bkonyi does this sounds familiar? I looked through the vm developer code but didn't catch what jacob is referencing.

File: packages/devtools_app/lib/src/error_badge_manager.dart (hunk 1 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/error_badge_manager.dart:39]: Yeah we should. Will handle this in a follow up, along with the `Flutter.error` events.

**Comment by kenzieschmoll** [comment]
> As a drive by comment on the look, did you try badging the lower right of the tab text? It might be alright to overlap the tab title (it looks good as is, but I was curious about how other options might look).

This is what it would look like in the bottom left. IMO I think the top right is a little better because it doesn't crowd the blue selection indicator at the bottom of the tab
Screen Shot 2020-12-18 at 10 11 12 AM

File: packages/devtools_app/lib/src/error_badge_manager.dart (hunk 2 above)
Comment (bkonyi) [reply packages/devtools_app/lib/src/error_badge_manager.dart:43]: Yup, you're looking for `FixedValueListenable` in `devtools_app/lib/src/listenable.dart`.

**Pull request merged**